spec/consensus/proposer-based-timestamp/pbts-sysmodel_002_draft.md

```diff
@@ -79,116 +81,163 @@ defining a lower bound, a *minimum time* that a correct process assigns to propo
+- [Agreement] No two correct processes decide on different values `v`.
+
+This implies that no two correct processes decide on different proposal times
+`v.time`.
 
 #### **[PBTS-INV-VALID.0]**
 
-[Validity] If a correct process decides on value `v`,
-then `v` satisfies a predefined `valid` predicate.
+- [Validity] If a correct process decides on value `v`, then `v` satisfies a
+  predefined `valid` predicate.
+
+With respect to PBTS, the `valid` predicate requires proposal times to be
+[monotonic](./pbts-algorithm_002_draft.md#time-monotonicity) over heights of
+consensus:
+
+##### **[PBTS-INV-MONOTONICITY.0]**
+
+- If a correct process decides on value `v` at the height `h` of consensus,
+  thus setting `decision[h] = v`, then `v.time > decision[h'].time` for all
+  previous heights `h' < h`.
+
+The monotonicity of proposal times, and external validity in general,
+implicitly assumes that heights of consensus are executed in order.
 
 #### **[PBTS-INV-TIMELY.0]**
 
-[Time-Validity] If a correct process decides on value `v`,
-then the associated proposal time `v.time` satisfies a predefined `timely` predicate.
+- [Time-Validity] If a correct process decides on value `v`, then the proposal
+  time `v.time` was considered `timely` by at least one correct process.
+
+PBTS introduces a `timely` predicate that restricts the allowed decisions based
+on the proposal time `v.time` associated with a proposed value `v`.
+As as synchronous predicate, the time at which it is evaluated impacts on
+whether a process accepts of reject a value based on its proposal time.
+For this reason, the Time-Validity property refers to the previous evaluation
+of the `timely` predicate, detailed in the following.
+
+## Timely proposals
+
+For PBTS, a `proposal` is a tuple `(v, v.time, v.round)`, where:
 
-> Both [Validity] and [Time-Validity] must be observed even if up to `2f` validators are faulty.
+- `v` is the proposed value;
+- `v.time` is the associated proposal time;
+- `v.round` is the round at which `v` was first proposed.
 
-### Timely proposals
+We introduce this definition of proposal because  a value `v` and its
+associated proposal time `v.time` can be proposed in multiple rounds, but the
+evaluation of the `timely` predicate is only relevant at round `v.round`.
+
+> Considering the algorithm in the [arXiv paper][arXiv], a new proposal is
+> produced by a proposer `p` when its `validValue_p` is unset.
+> The produced proposal thus have `v.round = round_p` and it is broadcast in a
+> Proposal message of with the `validRound` field set to `-1`.
+
+#### **[PBTS-PROPOSAL-RECEPTION.0]**
 
 The `timely` predicate is evaluated when a process receives a proposal.
-Let `now_p` be time a process `p` reads from its local clock when `p` receives a proposal.
-Let `v` be the proposed value and `v.time` the proposal time.
-The proposal is considered `timely` by `p` if:
+More precisely, let `p` be a correct process:
```

Comment:
Should this section mention `vr = -1` for this logic to be evaluated?

Reply:
In the TLA+ specification we save the proposal receive time for all Proposals, in followed the same approach here. 

But in fact this time is only relevant when the proposal round matches the proposal's message round, which is the same as saying that valid round = -1.